### DISCUSSION TITLE
Support for commercial DBs

### DISCUSSION THREAD

**Discussion by moshemarciano** [ent/ent#2781] [General]
Hi,

great project!

I don't see support for Microsoft SQL server and Oracle, is that a deliberate decision?

I think these are widely used in enterprise apps and limits where ent can be used.

Any plans to support it soon?

**Comment by a8m** [comment]
Hey @moshemarciano 👋 

Ent uses standard SQL for 99% of its statements, so that means supporting a new SQL database in the runtime code shouldn't be that hard. The complex part is the migration/schema management layer. Fortunately,  Ent uses [Atlas](https://github.com/ariga/atlas) for schema management (inspection, diffing, migration, etc), and we plan to add support for the 2 databases you mentioned in the upcoming months both in Atlas and Ent.

If you have a specific use case you would like to talk about, I'm available on our [Discord server](https://discord.gg/qZmPgTE6RX) 🙏

**Reply by moshemarciano** [reply]
thanks Ariel (@a8m ) 

That's great news

my use case is simple, looking for a db framework to rely on when dealing with PostgressSQL, SQL Server and Oracle because of existing install base and backwards compatibility.  

I guess we can start app building with Postgress support and test on the other two when they are supported in a few months.

**Reply by moshemarciano** [reply]
@a8m Any news on that? I don't see the commercial DBs in the documentation

**Reply by a8m** [reply]
Microsoft SQL Server was added to [Atlas](https://atlasgo.io/cloud/beta-drivers) as a beta driver (the others will be released soon). We (Ariga) plan to provide plugins for the different commercial drivers supported by Atlas soon.

**Reply by moshemarciano** [reply]
Hi @a8m I see Microsoft SQL Server mentioned in Atlas but not in Ent and none of them still specify Oracle support. As I need to still support Oracle from time to time as well as Postgres in one framework I am still on fence with this.

Any news you can share?

**Comment by matanyall** [comment]
Sorry for resurrecting an old thread, but I'd really love to have sql server support in ent, so I took a crack at it here: https://github.com/ent/ent/pull/4474

I need the sql server schema for testing migrations and the integration tests, will ping in the discord
